src/Generating/EndGen.cpp

```diff
@@ -133,17 +133,17 @@ void cEndGen::GenShape(cChunkCoords a_ChunkCoords, cChunkDesc::Shape & a_Shape)
 				// The distance from spawn is so big we don't need to calculate the max height anymore.
 				// In fact, if we don't cut it off somewhere there is a chance the maxheight gets too big which
 				// can cause corrupted looking terrain.
-				maxHeightLimit = static_cast<float>(cChunkDef::Height);
+				maxHeightLimit = static_cast<float>(cChunkDef::UpperLimit);
 			}
 			else
 			{
 				// Create a void between the main island and the other island using the formula 'x^3 - 3 * x' where x is distance from spawn.
 				double pow = std::pow((distanceFromSpawn - m_MainIslandSize) / m_MainIslandSize, 3);
 				double mult = 3 * ((distanceFromSpawn - m_MainIslandSize) / m_MainIslandSize);
-				maxHeightLimit = Clamp((pow - mult) * 100 + static_cast<double>(voidOffset) * m_VoidOffsetNoiseMultiplier, 0.0, static_cast<double>(cChunkDef::Height));
+				maxHeightLimit = Clamp((pow - mult) * 100 + static_cast<double>(voidOffset) * m_VoidOffsetNoiseMultiplier, static_cast<double>(cChunkDef::LowerLimit), static_cast<double>(cChunkDef::UpperLimit));
 			}
-			int maxHeight = static_cast<int>(Clamp(m_BaseHeight + static_cast<double>(noise) * m_TerrainTopMultiplier, 0.0, maxHeightLimit));
-			int minHeight = static_cast<int>(Clamp(m_BaseHeight - static_cast<double>(noise) * m_TerrainBottomMultiplier, 0.0, static_cast<double>(cChunkDef::Height)));
+			int maxHeight = static_cast<int>(Clamp(m_BaseHeight + static_cast<double>(noise) * m_TerrainTopMultiplier, static_cast<double>(cChunkDef::LowerLimit), maxHeightLimit));
+			int minHeight = static_cast<int>(Clamp(m_BaseHeight - static_cast<double>(noise) * m_TerrainBottomMultiplier, static_cast<double>(cChunkDef::LowerLimit), static_cast<double>(cChunkDef::UpperLimit)));
 
 			for (int y = minHeight; y < maxHeight; y++)
 			{
```

Comment:
(I can't comment on lines too much away from the diff)
Line 150 needs to be rewritten using `cChunkDef` constants.